### PR TITLE
 Clarify Activity parent in links example (readme)

### PR DESCRIPTION
**Pull request by lmolkova**
Server activity had parent equal to one of the links - this is confusing, changed to default context (new trace) as likely this activity has no parent.

### CODE REVIEW

File: src/OpenTelemetry.Api/README.md [hunk 1]
```diff
@@ -270,13 +270,17 @@ OpenTelemetry samplers chose not to sample this activity.
...
-        "ActivityName",
+        "ActivityWithLinks",
         ActivityKind.Server,
-        "00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-01",
+        default(ActivityContext), // this creates Activity without parent
         initialTags,
         activityLinks);
     ```
 
+    In case you have a parent and it's remote specify `traceaparent` string
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/README.md:280]: I'd rephrase to avoid confusion.
In case you have a parent -> In case the activity has a parent

File: src/OpenTelemetry.Api/README.md [hunk 2]
```diff
@@ -270,13 +270,17 @@ OpenTelemetry samplers chose not to sample this activity.
...
+    instead of the default context, or if you have parent Activity,
+    use `Activity.Current.Context` (check for null!).
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/README.md:282]: can we always ask to pass `ActivityContext` instead of `traceparent` string for remote vs actual context if not remote? Or do you want to wait until `ActivityContext` adds `IsRemote`?

Comment (cijothomas) [inline src/OpenTelemetry.Api/README.md:282]: (check for null!) -> lets show the exact code doing it.

File: src/OpenTelemetry.Api/README.md (hunk 1 above)
Comment (cijothomas) [inline src/OpenTelemetry.Api/README.md:280]: typoe - `traceaparent` to `traceparent`

Comment (reyang) [inline src/OpenTelemetry.Api/README.md:280]: I don't know if `traceparent` is a familiar term for the reader, consider adding a reference link.

File: src/OpenTelemetry.Api/README.md (hunk 2 above)
Comment (lmolkova) [reply src/OpenTelemetry.Api/README.md:282]: we happen to have overloads that take string parentId and ActivityContext. I guess if/once https://github.com/dotnet/runtime/issues/40046 is in (`public static bool TryParse(string w3cId, string? traceState, out ActivityContext context)`) we should always recommend people to construct parent context.
Do you think I should remove `traceparent` string for now and update the doc once  we have `TryParse`?

Comment (cijothomas) [reply src/OpenTelemetry.Api/README.md]: The overload taking parentId string was originally intended for Hierrachical Id format. It can be used in W3C format as well, but if the string is W3C, then it can be converted to ActivityContext and passed. The new `TryParse` would help get the ActivityContext.

I think its best to only advertise the ActivityContext overload.
